src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -42,105 +47,215 @@
+        .put("userid", UUID.randomUUID().toString());
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(tableName), 1, 1,
+        TimeUnit.MINUTES.toMillis(2));
+
+    KafkaConsumer<byte[], byte[]> consumer =
+        connect.kafka().createConsumerAndSubscribeTo(Collections.emptyMap(), DLQ_TOPIC_NAME);
+
+    ConsumerRecords<byte[], byte[]> records =
+        consumer.poll(Duration.ofMillis(CONSUME_MAX_DURATION_MS));
+
+    assertEquals(1, records.count());
+  }
+
+  @Test
+  public void testWriteToTableWithUuidColumn() throws Exception {
     createTableWithUuidColumns();
-    startTask();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
     final Schema schema = SchemaBuilder.struct().name("com.example.Person")
                                        .field("firstname", Schema.STRING_SCHEMA)
                                        .field("lastname", Schema.STRING_SCHEMA)
                                        .field("jsonid", Schema.STRING_SCHEMA)
                                        .field("userid", Schema.STRING_SCHEMA)
                                        .build();
-    UUID uuid = UUID.randomUUID();
-    String jsonid = "5";
     final Struct struct = new Struct(schema)
         .put("firstname", "Christina")
         .put("lastname", "Brams")
-        .put("jsonid", jsonid)
-        .put("userid", uuid.toString());
-    task.put(Collections.singleton(new SinkRecord(tableName, 1, null, null, schema, struct, 1)));
+        .put("jsonid", "5")
+        .put("userid", UUID.randomUUID().toString());
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(tableName), 1, 1,
+        TimeUnit.MINUTES.toMillis(2));
+
     try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
       try (Statement s = c.createStatement()) {
         try (ResultSet rs = s.executeQuery("SELECT * FROM " + tableName)) {
-          assertTrue(rs.next());
-          assertEquals(struct.getString("firstname"), rs.getString("firstname"));
-          assertEquals(struct.getString("lastname"), rs.getString("lastname"));
-          assertEquals(struct.getString("jsonid"), rs.getString("jsonid"));
-          assertEquals(struct.getString("userid"), rs.getString("userid"));
+          assertTrue(rs.next()
+              && struct.getString("firstname").equals(rs.getString("firstname"))
+              && struct.getString("lastname").equals(rs.getString("lastname"))
+              && struct.getString("jsonid").equals(rs.getString("jsonid"))
+              && struct.getString("userid").equals(rs.getString("userid")));
```

Comment:
I think we should keep this as-is as the error messages from `assertEquals` are more informative than those from `assertTrue`, and we can more-easily distinguish which tested condition was not met by testing them via separate content-based assertions instead of a single chained boolean assertion.